### ISSUE TITLE
[Grid] Grid has invalid value of `aria-rowindex='0'`

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3032]
## Current behavior

All Grid variants except for "Grouping" have issue with numeration of table rows inside `tbody`. First `tr` in each `tbody` has `aria-rowindex='0'` but its value cannot be less than 1. 
Because of this a11y add-on reports an issue "ARIA attribute values valid".

<img width="623" height="500" alt="Image" src="https://github.com/user-attachments/assets/a216ed0a-878f-433c-85f0-790e208511d9" />

## Sugested solution

It's recommended to shift each value of table row `aria-rowindex` by 1. Then the first table row will have `aria-rowindex='1'` and each `aria-rowindex` values after that would also have proper numeration.
First `tr` inside `tbody` should look like: `<tr aria-rowindex="1" index="0" part="row odd">`

### 👤 Your name

Kacper

### 🧢 Your Product/Team

SLDS team
